sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/MessageUtils.java

```diff
@@ -369,4 +380,167 @@ public static <T> T describedToOrigin(DescribedType describedType) {
+        if (describedFilter.getDescriptor().equals(ServiceBusConstants.SQL_FILTER_NAME)
+            && describedFilter.getDescribed() instanceof Iterable) {
+            Iterator<Object> describedSqlFilter = ((Iterable<Object>) describedFilter.getDescribed()).iterator();
+            if (describedSqlFilter.hasNext()) {
+                return new SqlRuleFilter((String) describedSqlFilter.next());
+            }
+        } else if (describedFilter.getDescriptor().equals(ServiceBusConstants.CORRELATION_FILTER_NAME)
+            && describedFilter.getDescribed() instanceof Iterable) {
+            CorrelationRuleFilter correlationFilter = new CorrelationRuleFilter();
+            Iterator<Object> describedCorrelationFilter = ((Iterable<Object>) describedFilter.getDescribed()).iterator();
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setCorrelationId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setMessageId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setTo((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setReplyTo((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setLabel((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setSessionId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setReplyToSessionId((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                correlationFilter.setContentType((String) (describedCorrelationFilter.next()));
+            }
+            if (describedCorrelationFilter.hasNext()) {
+                Object properties = describedCorrelationFilter.next();
+                if (properties instanceof Map) {
+                    correlationFilter.getProperties().putAll((Map<String, ?>) properties);
+                }
+            }
+
+            return correlationFilter;
+        } else if (describedFilter.getDescriptor().equals(ServiceBusConstants.TRUE_FILTER_NAME)) {
+            return new TrueRuleFilter();
+        } else if (describedFilter.getDescriptor().equals(ServiceBusConstants.FALSE_FILTER_NAME)) {
+            return new FalseRuleFilter();
+        } else {
+            throw new UnsupportedOperationException("This client cannot support filter with descriptor: " + describedFilter.getDescriptor());
+        }
+
+        return null;
```

Comment:
Is returning null a normal condition? (if so, just making sure calling code has a null check

Reply:
Already has a null check in calling code.

return null won't reach if service response correctly.